newIDE/app/src/UI/Theme/index.js

```diff
@@ -1,23 +1,61 @@
 // @flow
+import {
+  select,
+  blend as Install,
+} from '../../Utils/Object';
+import { isLTR } from '../../Utils/i18n/language';
+import domain from './domain';
+import DefaultTheme, {
+  themeName as defaultThemeName,
+  type Theme,
+} from './DefaultTheme';
 import DarkTheme from './DarkTheme';
-import DefaultTheme from './DefaultTheme';
 import NordTheme from './NordTheme';
 import SolarizedDarkTheme from './SolarizedDarkTheme';
-import { type Theme } from './DefaultTheme';
 import './Global.css';
 
+type MuiTheme = $Exact<typeof DefaultTheme.muiTheme>;
+
+/**
+ * Enhances every Theme instance with a `muiTheme` property accessor.
+ *
+ * This ensures the `muiTheme` is correctly applied whenever language
+ * directionality is changed. Thus, based on `language-ltr`, one of
+ * two _distinct_ `muiTheme` instances is returned; these two instances
+ * are exactly alike (i.e. have the exact same properties).
+ */
+const install = new Install(({ gdevelopTheme, muiTheme }: Theme): Theme => {
+  const muiThemes: [MuiTheme, MuiTheme] = [muiTheme, { ...muiTheme }];
+  return {
+    gdevelopTheme,
+    get muiTheme(): MuiTheme {
+      const ltr = domain['language-ltr'] ?? true;
+      return select(muiThemes, ltr);
+    },
+  };
+});
+
 // To add a new theme:
 // * copy the folder of an existing one (DarkTheme for example),
 // * import it at the top of the file
 // * add it below:
-export const themes = {
-  'GDevelop default': DefaultTheme,
+export const themes = install({
+  [defaultThemeName]: DefaultTheme,
   Dark: DarkTheme,
   Nord: NordTheme,
   'Solarized Dark': SolarizedDarkTheme,
-};
+});
 
-export const getTheme = (themeName: string): Theme =>
-  themes[themeName] || themes['GDevelop default'];
+export function getTheme({ themeName, language }: {| themeName: string, language: string |}): Theme {
+  // This boolean value indicates if the defined directionality of `language` is left-to-right
+  domain['language-ltr'] = isLTR(language);
```

Comment:
Let's entirely avoid to store things in a global variable `domain`. It's not typed and the name is too generic to understand what can go inside.